src/DotNetBridge/NativeDataView.cs

```diff
@@ -200,7 +140,10 @@ public NativeDataView(IHostEnvironment env, DataSourceBlock* pdata)
                 }
 
                 _columns = columns.ToArray();
-                Schema = Schema.Create(new SchemaImpl(_columns));
+                var schemaBuilder = new SchemaBuilder();
+                foreach (var col in columns)
+                    schemaBuilder.AddColumn(col.Name, col.Type, col.Metadata);
```

Comment:
If each column contains a Schema.DetachedColumn instead of these three, you can use schemaBuilder.AddColumns(IEnumerable<Schema.DetachedColumn>) method to add all the columns in one call.

Reply:
Simplified.

---
In reply to: [250277096](https://github.com/Microsoft/NimbusML/pull/90#discussion_r250277096) [](ancestors = 250277096)